src/Compilers/CSharp/Portable/Syntax/SyntaxExtensions.cs

```diff
@@ -70,15 +70,18 @@ public static SyntaxToken NormalizeWhitespace(this SyntaxToken token, string ind
         /// </summary>
         internal static TypeSyntax Type(this DeclarationExpressionSyntax self)
         {
-            return self.Declaration.Type;
+            var component = (TypedVariableComponentSyntax)self.Declaration;
```

Comment:
The parser never builds nodes that would fail this cast.

---
In reply to: [73446523](https://github.com/dotnet/roslyn/pull/12917#discussion_r73446523) [](ancestors = 73446523)